gatsby/src/pages/__tests__/topics.test.tsx

```diff
@@ -0,0 +1,171 @@
+		});
+	});
+
+	describe("Page header", () => {
+		it("should render heading 1 with speciality name", () => {
+			expect(
+				renderResult.queryByText("Topics A to Z", {
+					selector: "h1",
+				})
+			).toBeInTheDocument();
+		});
+		it("should render lead paragraph", () => {
+			expect(
+				renderResult.queryByText(
+					"There are over 350 topics, with focus on the most common and significant presentations in primary care."
+				)
+			).toBeInTheDocument();
+		});
+	});
+
+	describe("Alphabet", () => {
+		let alphabetList: HTMLElement;
+		beforeEach(() => {
+			const { getByLabelText } = renderResult;
+			alphabetList = getByLabelText("Letters A to Z");
+		});
+
+		it("should render an ordered list labelled Letters A to Z", () => {
+			expect(alphabetList).toHaveProperty("tagName", "OL");
+		});
+
+		it("should render a-to-z ID on alphabet list", () => {
+			expect(alphabetList).toHaveProperty("id", "a-to-z");
+		});
+
+		it("should render a list item for each letter", () => {
+			expect(alphabetList.children).toHaveLength(26);
+		});
+
+		it("should render internal anchor for letter with topics", () => {
+			const { getByLabelText } = renderResult;
+
+			const letterAAnchor = getByLabelText("Letter 'A'");
+			expect(letterAAnchor).toHaveProperty("tagName", "A");
+			expect(letterAAnchor).toHaveAttribute("href", "#a");
+			expect(letterAAnchor).toHaveTextContent("A");
+
+			const letterCAnchor = getByLabelText("Letter 'C'");
+			expect(letterCAnchor).toHaveProperty("tagName", "A");
+			expect(letterCAnchor).toHaveAttribute("href", "#c");
+			expect(letterCAnchor).toHaveTextContent("C");
+		});
+
+		it("should not render anchor for letter with no topics", () => {
+			const { getByLabelText } = renderResult;
+			const letterQAnchor = getByLabelText("Letter 'Q' (no topics)");
+			expect(letterQAnchor).not.toHaveProperty("tagName", "A");
+			expect(letterQAnchor).toHaveTextContent("Q");
+		});
+	});
+
+	describe("A to Z", () => {
+		it("should render A to Z navigation element", () => {
+			const { getByLabelText } = renderResult;
+			expect(getByLabelText("Topics A to Z")).toHaveProperty("tagName", "NAV");
+		});
+
+		it("should render ordered list of letters that have at least 1 topic", () => {
+			const { getByLabelText } = renderResult;
+			const list = getByLabelText("Letters A to Z with matching topics");
+			expect(list).toHaveProperty("tagName", "OL");
+			expect(list.children).toHaveLength(2); // See test data above
+		});
+
+		it("should render heading 2 with id and label for each letter", () => {
+			const { getByLabelText } = renderResult;
+			const letterHeading = getByLabelText("Topics starting with 'A'");
+			expect(letterHeading).toHaveProperty("tagName", "H2");
+			expect(letterHeading).toHaveAttribute("id", "a");
+		});
+
+		it("should render list of topics labelled by the letter heading for each letter", () => {
+			const { getByLabelText } = renderResult;
+			expect(getByLabelText("C")).toHaveProperty("tagName", "OL");
```

Comment:
I was expecting this to by LI not OL as the letter C would be a list item no?

Reply:
Well it's a nested list of letters, with topics inside, like:

1. A
    1. Achilles
    2. Asthma
2. B
    1. Bakes'c cyst

And the topic list is an OL, because it's ordered alphabetically and it's labelled by the heading that's in the parent list. So it should be an OL.

Reply:
Good stuff, thanks for clearing that up